drivers/hmis/app/graphql/types/hmis_schema/has_clients.rb

```diff
@@ -0,0 +1,33 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Types
+  module HmisSchema
+    module HasClients
+      extend ActiveSupport::Concern
+
+      class_methods do
+        def clients_field(name = :clients, description = nil, type: [Types::HmisSchema::Client], **override_options, &block)
+          default_field_options = { type: type, null: false, description: description }
+          field_options = default_field_options.merge(override_options)
+          field(name, **field_options) do
+            argument :sort_field, Types::HmisSchema::ClientSortOption, required: false
+            argument :sort_direction, Types::HmisSchema::SortDirection, required: false
+            instance_eval(&block) if block_given?
+          end
+        end
+      end
+
+      def resolve_clients(scope = object.clients, sort_field: :LastName, sort_direction: :asc, _user: current_user, no_sort: false)
+        clients_scope = scope
+        clients_scope = clients_scope.order(sort_field => sort_direction) unless no_sort
```

Comment:
`.order(sort_field => sort_direction)` probably won't hold up in the long run.  We've found we're often sorting on things that aren't on the Client table.  Just a note for the future, but we're probably going to want to pass in a `sort_method` which would look something like `arel_table[:LastName].asc` where `arel_table` is something like `c_t` which is defined in ArelHelper.

Reply:
Yeah, makes sense. `sort_field` comes from an enum, so we do control what values are allowed for that. Regardless, I agree that this probably isn't going to work forever. When the time comes we can (and should) either expand the sorting logic in this concern or defer to some other abstraction to handle it.

Reply:
I'm already hitting that with implementing project sort. We want to sort projects by Organization name and Project name. WIP PR here https://github.com/greenriver/hmis-warehouse/pull/2335 but I'm not sure how to best implement the mapping of Enums to their sort functions. @ZachCMP sounds like you had some ideas there –  can we take a look together?

Reply:
@gigxz I think we ought not put the sorting logic in the enum. My thought is that we put it elsewhere, maybe in the model, maybe in its own utility class or something. What we could do is enumerate the sorting options there and then generate the GraphQL enum from that. That feels more appropriate to me.